source/simulator/core.cc

```diff
@@ -2056,26 +2102,61 @@ namespace aspect
         initialize_tracers ();
 
         computing_timer.exit_section();
+
+        if (parameters.nonlinear_solver == NonlinearSolver::Binary_input)
+        {
+          const std::string filename = parameters.binary_data_directory + "/fields-" +
+                                       Utilities::int_to_string(parameters.binary_data_ts_number, 5) + ".bin";
+          std::ifstream input_stream (filename);
+          AssertThrow(input_stream,
+                      ExcMessage(std::string("Failed to read the file, " + filename +
+                                             " for the time step number, " + Utilities::int_to_string(parameters.binary_data_ts_number) + " using Nonlinear solver mode \"Binary_input\".")));
+          Utilities::BinaryInputFields input_fields;
+          boost::archive::binary_iarchive ia(input_stream);
+          ia >> input_fields;
+          time = input_fields.time;
+          time_step = input_fields.time_step;
+          timestep_number = input_fields.timestep_number;
+          old_time_step = input_fields.old_time_step;
+          input_stream.close();
+        }
       }
 
     // start the principal loop over time steps:
-    do
-      {
-        start_timestep ();
+    do {
+      start_timestep();
 
-        // then do the core work: assemble systems and solve
-        solve_timestep ();
+      // then do the core work: assemble systems and solve
+      solve_timestep();
 
-        pcout << std::endl;
+      pcout << std::endl;
 
-        // update the time step size
-        // for now the bool (convection/conduction dominated)
-        // returned by compute_time_step is unused, will be
-        // added to statistics later
+      // update the time step size
+      // for now the bool (convection/conduction dominated)
+      // returned by compute_time_step is unused, will be
+      // added to statistics later
+      if (parameters.nonlinear_solver != NonlinearSolver::Binary_input) {
         old_time_step = time_step;
-        time_step = std::min (compute_time_step().first,
-                              parameters.maximum_time_step);
+        time_step = std::min(compute_time_step().first,
+                             parameters.maximum_time_step);
         time_step = termination_manager.check_for_last_time_step(time_step);
+      }
+      else {
+        const std::string filename = parameters.binary_data_directory + "/fields-" +
```

Comment:
Did you check that you need to do this? If the solution is identical, the computed timestep size should be the same too, right?